nixos/modules/system/activation/switch-to-configuration.pl

```diff
@@ -104,14 +106,27 @@ sub getActiveUnits {
     return $res;
 }
 
+# Returns whether a systemd unit is active
+sub unit_is_active {
+    my ($unit_name) = @_;
+
+    my $mgr = Net::DBus->system->get_service('org.freedesktop.systemd1')->get_object('/org/freedesktop/systemd1');
+    my $units = $mgr->ListUnitsByNames([$unit_name]);
+    if (scalar(@{$units}) == 0) {
+        return 0;
+    }
+    my $active_state = @{@{$units}[0]}[3]; ## no critic (ValuesAndExpressions::ProhibitMagicNumbers)
```

Comment:
This could probably be written in a simpler way:

```suggestion
    my $active_state = $units->[0]->[3]; 
```